Venv "can't open file file" named cache flake

Seen in CI on 2.11:

```
03:44:10.89 [ERROR] Completed: Lint with autoflake - autoflake failed (exit code 2).
/home/runner/.cache/pants/named_caches/pex_root/venvs/s/e4a8d8cc/venv/bin/python3.7: can't open file '/home/runner/.cache/pants/named_caches/pex_root/venvs/76dfc7bf5b19524ee737148e3049712704fec4e9/2703f860957c2efe7f4d087e27e538da0891bc96/pex': [Errno 2] No such file or directory
```

https://github.com/pantsbuild/pants/runs/5328703987?check_suite_focus=true#step:10:271

See https://pantsbuild.slack.com/archives/C0D7TNJHL/p1645761742140559 for discussion.

--- a/src/python/pants/backend/python/util_rules/pex.py
+++ b/src/python/pants/backend/python/util_rules/pex.py
@@ -749,8 +749,7 @@ class VenvScriptWriter:
 
             # If the seeded venv has been removed from the PEX_ROOT, we re-seed from the original
             # `--venv` mode PEX file.
-            if [ ! -e "${{target_venv_executable}}" ]; then
-                rm -rf "${{venv_dir}}" || true
+            if [ ! -e "${{venv_dir}}" ]; then
                 PEX_INTERPRETER=1 ${{execute_pex_args}} -c ''
             fi
 

